cloud/kubernetes/cockroachdb-client-secure.yaml

```diff
@@ -0,0 +1,51 @@
+apiVersion: apps/v1beta1
+kind: Deployment
+metadata:
+  name: cockroachdb-client-secure
+spec:
+  replicas: 1
+  template:
+    metadata:
+      labels:
+        app: cockroachdb
+    spec:
+      volumes:
+      - name: client-certs
+        emptyDir: {}
+
+      initContainers:
+      # The init-certs container sends a certificate signing request to the
+      # kubernetes cluster.
+      # You can see pending requests using: kubectl get csr
+      # CSRs can be approved using:         kubectl certificate approve <csr name>
+      #
+      # In addition to the client certificate and key, the init-certs entrypoint will symlink
+      # the cluster CA to the certs directory.
+      - name: init-certs
+        image: cockroachdb/cockroach-k8s-request-cert:0.2
+        imagePullPolicy: IfNotPresent
+        command:
+        - "/bin/ash"
+        - "-ecx"
+        - "/request-cert -namespace=${POD_NAMESPACE} -certs-dir=/cockroach-certs -type=client -user=root -symlink-ca-from=/var/run/secrets/kubernetes.io/serviceaccount/ca.crt"
+        env:
+        - name: POD_NAMESPACE
+          valueFrom:
+            fieldRef:
+              fieldPath: metadata.namespace
+        volumeMounts:
+        - name: client-certs
+          mountPath: /cockroach-certs
+
+      containers:
+      - name: cockroachdb-client
+        image: cockroachdb/cockroach:v1.1.2
+        imagePullPolicy: IfNotPresent
+        volumeMounts:
+        - name: client-certs
+          mountPath: /cockroach-certs
+        # Keep a pod open indefinitely so kubectl exec can be used to get a shell to it
+        # and run cockroach client commands, such as cockroach sql, cockroach node status, etc.
+        command:
```

Comment:
@a-robinson: do you know if there's a way to do without the `sleep 999...` hack?

Reply:
The container has to have some sort of command, and we don't want that command to waste machine resources. If you know of a less ugly command, we can switch to that, but we do need to tell it to do something AFAIK.